plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraTable.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.cassandra;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.testing.sql.SqlExecutor;
+import io.trino.testing.sql.TestTable;
+
+import java.util.List;
+import java.util.function.Function;
+
+import static com.google.common.base.Verify.verify;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.lang.String.format;
+import static java.lang.String.join;
+import static java.util.stream.Collectors.joining;
+
+public class TestCassandraTable
+        extends TestTable
+{
+    private final String tableName;
+
+    // Make sure the `namePrefix` in the following constructors does not exceed 38 characters,
+    // because Cassandra table names are limited to 48 characters in size, so we can only use
+    // a maximum of 38 characters without the suffix.
+    // https://cassandra.apache.org/doc/trunk/cassandra/cql/ddl.html#common-definitions
+    //
+    // NOTE: "38 characters" may be changed in the future, please refer to
+    // `48 - TestTable.RANDOM_SUFFIX_LENGTH` for the exact value.
+
+    public TestCassandraTable(
+            SqlExecutor sqlExecutor,
+            CassandraServer server,
+            String keyspace,
+            String namePrefix,
+            List<ColumnDefinition> columnDefinitions,
+            List<String> rowsToInsert)
```

Comment:
As I commented above, I think @ebyhr's suggestion is better, because if we extend the `TestTable`, in the future the table name suffix length may affect the existing tests, so it is better to fix the table name suffix length by implementing the `AutoCloseable`, and it really can't reduce much duplicate code in this change by extending it.

cc @hashhar